codeclimate_test_reporter/components/file_coverage.py

```diff
@@ -18,6 +22,7 @@ def payload(self):
         }
 
     def __process(self):
+        self.file_body = open(self.__filename(), "r", encoding="utf-8-sig").read()
```

Comment:
To answer your question, I don't think we'll encounter paths through `__process` that wouldn't have to read the file. The file contents are used to compute the `blob` as well as to seed the `coverage` array.